app/core/Engine/controllers/rewards-controller/RewardsController.ts

```
@@ -267,15 +294,57 @@ export class RewardsController extends BaseController<
           state.lastAuthenticatedAccount = address;
           state.lastAuthTime = Date.now();
         });
+
+        // Publish event for failed authentication (not opted in)
+        this.messagingSystem.publish(
+          'RewardsController:selectedAccountAuthChange',
+          {
+            account,
+            subscriptionId: null,
+            error: false,
+          },
+        );
       } else {
         Logger.log(
           'RewardsController: Silent auth failed:',
           error instanceof Error ? error.message : String(error),
         );
+
+        // Publish event for failed authentication (other errors)
+        this.messagingSystem.publish(
+          'RewardsController:selectedAccountAuthChange',
+          {
+            account,
+            subscriptionId: null,
+            error: true,
+          },
+        );
       }
       // For other errors, we don't update the state to allow retries
     } finally {
       this.#isProcessingSilentAuth = false;
     }
   }
+
+  /**
+   * Estimate points for a given activity.
+   * @param estimatePointsDto - The estimate points request data.
+   * @returns The estimated points response.
+   */
+  async estimatePoints(
+    estimatePointsDto: EstimatePointsDto,
+  ): Promise<EstimatedPointsDto> {
+    try {
+      return await this.messagingSystem.call(
```

If I’m not mistaken, neither the Perps nor Swap teams should show rewards when a user is accessing the features from the UK. Can you throw an error before the return statement if the user’s location is detected to be in that region? There's [an endpoint](https://on-ramp.api.cx.metamask.io/geolocation) that returns user's location.